packages/core/src/external/aws/lambda-logic/modify-resources.ts

```diff
@@ -0,0 +1,599 @@
+    status,
+    reading: getUniqueDisplaysString(res.code),
+    value: res.valueCodeableConcept
+      ? getUniqueDisplaysString(res.valueCodeableConcept)
+      : getQuantityString(res.valueQuantity),
+    interpretation: getUniqueDisplaysString(res.interpretation),
+    referenceRange: refRange,
+  };
+}
+
+export type SlimMedication = Omit<Medication, "name"> & {
+  name?: string | undefined;
+  reference?: Record<string, string>;
+};
+
+function slimMedication(res: Medication): SlimMedication {
+  const updRes = cloneDeep(res);
+  delete updRes.code;
+  return {
+    ...updRes,
+    name: getUniqueDisplaysString(res.code),
+  };
+}
+
+export type SlimMedicationRequest = Omit<MedicationRequest, "requester"> & {
+  reference?: Record<string, Partial<SlimMedication>>;
+};
+
+function slimMedicationRequest(res: MedicationRequest): SlimMedicationRequest {
+  const updRes = cloneDeep(res);
+  delete updRes.requester;
+  return {
+    ...updRes,
+  };
+}
+
+export type SlimMedicationStatement = Omit<MedicationStatement, "dosage"> & {
+  dosages?: Dosage[] | undefined;
+  reference?: Record<string, Partial<SlimMedication>>;
+};
+
+type Dosage = {
+  dose: string | undefined;
+  route: string | undefined;
+};
+
+function slimMedicationStatement(res: MedicationStatement): SlimMedicationStatement {
+  const dosages = res.dosage?.flatMap(dosage => {
+    const dose = getQuantityString(dosage.doseAndRate?.[0]?.doseQuantity);
+    const route = getUniqueDisplaysString(dosage.route);
+    if (!dose && !route) return [];
+    return { dose, route };
+  });
+
+  const updRes = cloneDeep(res);
+  delete updRes.dosage;
+
+  return {
+    ...updRes,
+    dosages: dosages?.length ? dosages : undefined,
+  };
+}
+
+export type SlimMedicationAdministration = Omit<
+  MedicationAdministration,
+  "dose" | "route" | "dosage"
+> & {
+  dose?: string | undefined;
+  route?: string | undefined;
+  reference?: Record<string, Partial<SlimMedication>>;
+};
+
+function slimMedicationAdministration(res: MedicationAdministration): SlimMedicationAdministration {
+  const updRes = cloneDeep(res);
+  const dose = getQuantityString(res.dosage?.dose);
+  const route = getUniqueDisplaysString(res.dosage?.route);
+  delete updRes.dosage;
+
+  return {
+    ...updRes,
+    dose,
+    route,
+  };
+}
+
+export type SlimCondition = Omit<Condition, "name" | "category" | "clinicalStatus" | "code"> & {
+  name?: string | undefined;
+  category?: string | undefined;
+  clinicalStatus?: string | undefined;
+  reference?: Record<string, string | Partial<SlimPractitioner>>;
+};
+
+function slimCondition(res: Condition): SlimCondition {
+  const updRes = cloneDeep(res);
+  delete updRes.code;
+
+  return {
+    ...updRes,
+    name: getUniqueDisplaysString(res.code),
+    category: getUniqueDisplaysString(res.category),
+    clinicalStatus: isUselessStatus(getUniqueDisplaysString(res.clinicalStatus))
+      ? undefined
+      : getUniqueDisplaysString(res.clinicalStatus),
```

Comment:
ditto: don't use the original constant after it's cloned, this is dangerous - we cloned it in the first place for a reson.

Reply:
fixed